imageio/core/request.py

```diff
@@ -534,10 +532,7 @@ def finish(self):
 
         # Remove temp file
         if self._filename_local:
-            try:
-                os.remove(self._filename_local)
-            except Exception:  # pragma: no cover
-                pass
+            os.remove(self._filename_local)
```

Comment:
I'd keep the try-except. It should very rare, but if the remove fails for some reason, its tough luck, but lets not error our because of it.

Reply:
Hm, I think we really should error out because of this. All the situations I can come up with in which the call fails are undesirable, and the user should be informed about it:

1) The file disappears somewhere between the previous read and this `remove`. In this case, ImageIO should complain, because the user is messing with our internals in the wrong way. Alternatively, the user has a _very_ unreliable file system and likely bigger problems than ImageIO failing.

2) The file is still in use and can not be deleted. In this case, we have a bug in our codebase and would like to know about it. Eating the exception won't help.

3) ImageIO does not have permission to delete the file.  This is odd, because we created the file there earlier and (unless I am mistaken) create and delete permission comes as a pair. Either way, this is likely a bug in ImageIO as well, because we should avoid writing temp files to places we can't clean up.

That's the three scenarios I could come up with and in all of these, I'd rather fail noisily than end up with a potentially hard-to-trace bug.

Reply:
In my experience its hard to estimate what can go wrong if you talk about file systems. I've been bitten by this in other projects. But you're right that swallowing the exceptions may also hide internal bugs. What about we display a warning instead of raising an error?

Reply:
Yes, we could go with a warning instead of an error. 

Do you have a good idea on how to set up a test that would trigger this case? If we keep the except block, we might as well cover it with a test.

Reply:
Maybe this?:
* Create request object and make it use the `_filename_local`.
* Delete that file in the test.
* Call request.finish()

Reply:
@almarklein  This would fail earlier on in the same function (line 509) because we can't read from it. This is what caused me to remove the try/except block initially.

What we could possibly try a weird hack like:

- Create request object and make it use the _filename_local
- change the permissions on the directory to read-only (is this from inside python?)
- Call request.finish()
- set the permissions on the directory to read/write again

Reply:
Sometimes writing a test is not worth the effort. I would also be perfectly ok with:
* Open an interactive session
* Create request object and make it use the _filename_local
* Manually make that file read-only
* Call request.finish()
* Check that it does what you expect
* Add a comment that this piece of code was only tested manually.

Reply:
@almarklein Sure. I will see if I can work out how to tell codecove to ignore the except branch. 

From the top of my head, I could ignore any warnings that we raise, but that seems like too coarse a solution. Maybe there is some line magic to ignore a scope...